packages/rest/test/unit/rest-server/rest-server.test.ts

```diff
@@ -95,18 +98,23 @@ describe('Application', () => {
   describe('configuration', () => {
     it('uses http port 3000 by default', () => {
       const app = new Application();
-      expect(app.getSync(CoreBindings.HTTP_PORT)).to.equal(3000);
+      const server = new RestServer(app, {});
+      expect(server.getSync(RestBindings.PORT)).to.equal(3000);
     });
 
     it('can set port 0', () => {
-      const app = new Application({http: {port: 0}});
-      expect(app.getSync(CoreBindings.HTTP_PORT)).to.equal(0);
+      const app = new Application();
+      const server = new RestServer(app, {http: {port: 0}});
+      expect(server.getSync(RestBindings.PORT)).to.equal(0);
     });
   });
 
   function givenRequestContext(rootContext: Context = new Application()) {
-    const requestContext = new Context(rootContext);
-    requestContext.bind(CoreBindings.Http.CONTEXT).to(requestContext);
+    const app = rootContext as Application;
```

Comment:
Well, if this helper  requires `rootContext` to be an application, then you should say that in the argument signature.

```ts
function givenRequestContext(app: Application = new Application()) {
  app.server(RestServer);
  // ...
}
```

However, I feel this test is no longer a pure unit-test in such case, since we are bringing too much other dependencies into play.

Is it a necessary requirement that `RestServer` requires an application and that mere root `Context` is not enough?

**As a rule, unit tests shouldn't be using IoC container (context) to resolve dependencies. Dependencies should be resolved explicitly, using stub classes in most (if not all) cases.**